30413/Bogdan Emilian Sandu/Lab03/Extra/main.c

```diff
@@ -0,0 +1,139 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef struct node
+{
+    int val;
+    struct node *next;
+}NodeT;
+
+typedef struct
+{
+    NodeT *first;
+    NodeT *last;
+}ListT;
+
+NodeT *createNode(int val)
+{
+    NodeT *p;
+    p=(NodeT*)malloc(sizeof(NodeT));
+    if(p==NULL) exit(1);
+    p->val=val;
+    p->next=NULL;
+    return p;
+}
+
+ListT *createList()
+{
+    ListT *listPtr;
+    listPtr=(ListT*)malloc(sizeof(ListT));
+    if(listPtr==NULL) exit(1);
+    listPtr->first=listPtr->last=NULL;
+    return listPtr;
+}
+void insertFront(NodeT *p,ListT *listPtr)
+{
+    if(listPtr->first==NULL)
+    {
+        listPtr->first=listPtr->last=p;
+        p->next=NULL;
+    }
+    else
+    {
+        p->next=listPtr->first;
+        listPtr->first=p;
+    }
+}
+NodeT *interNode(ListT *listPtr1, ListT *listPtr2)//functia pentru intersectie
+{
+    NodeT *q1,*q2;
+    if(listPtr1->first==NULL || listPtr2->first==NULL) return NULL;
+    q1=listPtr1->first;
+    while(q1!=NULL)
+    {
+     q2=listPtr2->first;
+     while(q2!=NULL)
+     {
+       if(q2->val==q1->val) return q2;
+       q2=q2->next;
+     }
+     q1=q1->next;
+    }
+}
+void removeDups(ListT *listPtr)//functia pentru stergere a duplicatelor
```

Comment:
Ce complexitate in timp zici ca are functia asta? 

Complexitatea in timp este o functie care incearca sa estimeze variatia timpului de executie a unui program in functie de marimea inputului.

Reply:
Probabil n^2.

Reply:
Asa-i. Poti sa o faci in O(n logn)